examples/CampaignManagement/ValidateTextAd.php

```diff
@@ -0,0 +1,197 @@
+
+    /**
+     * Runs the example.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param int $customerId the customer ID
+     * @param int $adGroupId the ad group ID to validate the ad againts
+     */
+    public static function runExample(
+        GoogleAdsClient $googleAdsClient,
+        int $customerId,
+        int $adGroupId
+    ) {
+        // Creates the expanded text ad info.
+        $expandedTextAdInfo = new ExpandedTextAdInfo([
+            'description' => new StringValue(['value' => 'Luxury Cruise to Mars']),
+            'headline_part1' => new StringValue(['value' => 'Visit the Red Planet in style.']),
+            'headline_part2' => new StringValue(['value' => 'Low-gravity fun for everyone!!'])
+        ]);
+
+        // Sets the expanded text ad info on an ad.
+        $ad = new Ad([
+            'expanded_text_ad' => $expandedTextAdInfo,
+            'final_urls' => [new StringValue(['value' => 'http://www.example.com'])]
+        ]);
+
+        // Creates an ad group ad to hold the above ad.
+        $adGroupAd = new AdGroupAd([
+            'ad_group' => new StringValue([
+                'value' => ResourceNames::forAdGroup($customerId, $adGroupId)
+            ]),
+            // Optional: Set the status.
+            'status' => AdGroupAdStatus::PAUSED,
+            'ad' => $ad
+        ]);
+
+        // Creates the ad group ad operation.
+        $adGroupAdOperation = new AdGroupAdOperation();
+        $adGroupAdOperation->setCreate($adGroupAd);
+        $adGroupAdServiceClient = $googleAdsClient->getAdGroupAdServiceClient();
+
+        try {
+            // Try sending a mutate request to add the ad group ad.
+            $response = $adGroupAdServiceClient->mutateAdGroupAds(
+                $customerId,
+                [$adGroupAdOperation],
+                ['validateOnly' => true]
+            );
+
+            // Since validation is ON, result will be null.
+            printf("Expanded text ad validated successfully.%s", PHP_EOL);
+        } catch (GoogleAdsException $googleAdsException) {
+            // This block will be hit if there is a validation error from the server.
+            printf("There were validation error(s) while adding expanded text ad.%s", PHP_EOL);
+
+            $count = 1;
+            foreach ($googleAdsException->getGoogleAdsFailure()->getErrors() as $googleAdsError) {
+                /** @var GoogleAdsError $googleAdsError */
+                if ($googleAdsError->getErrorCode()->getPolicyFindingError() ==
+                    PolicyFindingError::POLICY_FINDING
+                    || $googleAdsError->getErrorCode()->getPolicyViolationError() ==
+                    PolicyViolationError::POLICY_ERROR) {
+                    // Only one of PolicyFindingDetails or PolicyViolationDetails will be
+                    // populated. PolicyViolationDetails is used by some ad formats, and
+                    // PolicyFindingDetails by others.
```

Comment:
In this example, I think we should scope down to only one class either PolicyViolationDetails or PolicyFindingDetails. You can just link to my [created guide for details](https://developers.google.com/google-ads/api/docs/policy-exemption/overview?hl=en).
As ETA only returns errors in PolicyFindingDetails, we should remove the part related to PolicyViolationErrors.

Reply:
Makes sense to me, I replicated what is in the [golden](https://github.com/googleads/google-ads-dotnet/blob/master/examples/CampaignManagement/ValidateTextAd.cs).

@AnashOommen : WDYT? Any chance to update the Golden soon (before other client libraries start porting)?

Reply:
I'll fix the golden and point to https://developers.google.com/google-ads/api/docs/policy-exemption/overview for more details.

Reply:
https://github.com/googleads/google-ads-dotnet/pull/148

Reply:
Thanks @AnashOommen , I updated too